perf(linter/no-restricted-types): skip running if config is empty

camc314 opened the pull request:
@connorshea i thought about doing this, https://github.com/oxc-project/oxc/pull/16276#discussion_r2573090747 But i decided the complexity wasn't worth it.

The only benefit we would get is managing to skip running this rule on some typescript files that have zero type names, however those are so few and far between, the added complexity of adding these custom checks, does not outweigh the perf behifit of implementing those checks.

connorshea commented on crates/oxc_linter/src/rules/typescript/no_restricted_types.rs:
```diff
@@ -255,7 +255,10 @@ impl Rule for NoRestrictedTypes {
+        if !ctx.source_type().is_typescript() || self.0.types.is_empty() {
+            return false;
+        }
+        true
```
I also prefer this

connorshea commented:
Only change req would be to add a test to the pass array to ensure the code works fine with an empty object like this:

```rs
("let f: any = true", Some(serde_json::json!([{ "types": {} }]))),
```

camc314 commented:
> Only change req would be to add a test to the pass array to ensure the code works fine with an empty object like this:
> 
> ```rust
> ("let f: any = true", Some(serde_json::json!([{ "types": {} }]))),
> ```

:+1: added

camc314 commented on crates/oxc_linter/src/rules/typescript/no_restricted_types.rs:
```diff
@@ -255,7 +255,10 @@ impl Rule for NoRestrictedTypes {
+        if !ctx.source_type().is_typescript() || self.0.types.is_empty() {
+            return false;
+        }
+        true
```
thanks, good eye

camc314 replied:
maybe it's a sign i should stop working on a saturday night 😂

connorshea reviewed (approved):
LGTM, thanks!

camc314 committed:
perf(linter/no-restricted-types): skip running if config is empty (#16278)

@connorshea i thought about doing this, https://github.com/oxc-project/oxc/pull/16276#discussion_r2573090747 But i decided the complexity wasn't worth it.

The only benefit we would get is managing to skip running this rule on some typescript files that have zero type names, however those are so few and far between, the added complexity of adding these custom checks, does not outweigh the perf behifit of implementing those checks.
